src/gui/qgsadvanceddigitizingcanvasitem.cpp

```diff
@@ -116,19 +129,19 @@ void QgsAdvancedDigitizingCanvasItem::paint( QPainter *painter )
     double a0, a;
     if ( mAdvancedDigitizingDockWidget->constraintAngle()->relative() && nPoints > 2 )
     {
-      a0 = std::atan2( -( prevPoint.y() - penulPoint.y() ), prevPoint.x() - penulPoint.x() );
+      a0 = std::atan2( -( prevPoint.y() - penulPoint.y() ), prevPoint.x() - penulPoint.x() ) + canvasRotationRad;
     }
     else
     {
-      a0 = 0;
+      a0 = canvasRotationRad;
     }
     if ( mAdvancedDigitizingDockWidget->constraintAngle()->isLocked() )
     {
       a = a0 - mAdvancedDigitizingDockWidget->constraintAngle()->value() * M_PI / 180;
     }
     else
     {
-      a = std::atan2( -( curPoint.y() - prevPoint.y() ), curPoint.x() - prevPoint.x() );
+      a = std::atan2( -( curPoint.y() - prevPoint.y() ), curPoint.x() - prevPoint.x() ) + canvasRotationRad;
```

Comment:
looking at the code, it seems a0 doesn't need to be shifted, and a simple `a += canvasRotationRad` could be added after the if block (at line 146).
That seems a bit more logic / easier to read.

Reply:
I've moved the statement for shifting `a` and `a0` - you're right, that is easier to understand.  `a0` still needs to be shifted, though, because it is used for drawing the circular arc later on.